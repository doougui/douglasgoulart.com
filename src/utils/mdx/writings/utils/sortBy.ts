import { Writing } from 'types/Writing';
import { parseToDate } from 'utils/time/parseToDate';
import { SortTypesKeys } from '../types';

function byNew(data: Writing[]) {
  return data.sort((a, b) => {
    const aAsDate = parseToDate(a.publishedAt);
    const bAsDate = parseToDate(b.publishedAt);
    return Number(bAsDate) - Number(aAsDate);
  });
}

function byName(data: Writing[]) {
  return data.sort((a, b) => (a.title > b.title ? 1 : -1));
}

export function sortBy(data: Writing[], type: SortTypesKeys): Writing[] {
  const sortTypes: { [key in SortTypesKeys]: () => Writing[] } = {
    new: () => byNew(data),
    name: () => byName(data),
  };

  return sortTypes[type]();
}
