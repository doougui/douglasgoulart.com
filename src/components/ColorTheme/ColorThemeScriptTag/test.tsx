import { render } from 'utils/tests';
import { ColorThemeScriptTag } from '.';

describe('<ColorThemeScriptTag />', () => {
  it('should render correctly', () => {
    const { container } = render(<ColorThemeScriptTag />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
