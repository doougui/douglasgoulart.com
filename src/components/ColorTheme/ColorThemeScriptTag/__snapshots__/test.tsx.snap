// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ColorThemeScriptTag /> should render correctly 1`] = `
<script>
  (function setColorsByTheme(storageKey, variableNameGetter) {
  /* istanbul ignore next */
  cov_d6phap09z().f[1]++;
  const colors =
  /* istanbul ignore next */
  (cov_d6phap09z().s[7]++, {"primary":{"light":"#107BEB","dark":"#107BEB"},"secondary":{"light":"#FDF9FC","dark":"#1C1C1C"},"tertiary":{"light":"#F3F2F3","dark":"#2D2D2F"},"background":{"light":"#FEFEFE","dark":"#151718"},"textDarker":{"light":"#C8C7CB","dark":"#504F57"},"textDark":{"light":"#858585","dark":"#888888"},"highlightText":{"light":"#EDEDEF","dark":"#EDEDEF"},"text":{"light":"#1A1523","dark":"#EDEDEF"},"success":{"light":"#26FF56","dark":"#26FF56"},"danger":{"light":"#FF4747","dark":"#FF4747"}});
  function getColorMode() {
    /* istanbul ignore next */
    cov_d6phap09z().f[2]++;
    const mql =
    /* istanbul ignore next */
    (cov_d6phap09z().s[8]++, window.matchMedia("(prefers-color-scheme: dark)"));
    const prefersDarkFromMQ =
    /* istanbul ignore next */
    (cov_d6phap09z().s[9]++, mql.matches);
    const defaultOS =
    /* istanbul ignore next */
    (cov_d6phap09z().s[10]++, prefersDarkFromMQ ?
    /* istanbul ignore next */
    (cov_d6phap09z().b[0][0]++, "dark") :
    /* istanbul ignore next */
    (cov_d6phap09z().b[0][1]++, "light"));
    const storagedMode =
    /* istanbul ignore next */
    (cov_d6phap09z().s[11]++, localStorage.getItem(storageKey));
    /* istanbul ignore next */
    cov_d6phap09z().s[12]++;
    if (!storagedMode) {
      /* istanbul ignore next */
      cov_d6phap09z().b[1][0]++;
      cov_d6phap09z().s[13]++;
      return defaultOS;
    } else
    /* istanbul ignore next */
    {
      cov_d6phap09z().b[1][1]++;
    }
    cov_d6phap09z().s[14]++;
    return storagedMode;
  }
  const root =
  /* istanbul ignore next */
  (cov_d6phap09z().s[15]++, document.documentElement);
  const mode =
  /* istanbul ignore next */
  (cov_d6phap09z().s[16]++, getColorMode());
  /* istanbul ignore next */
  cov_d6phap09z().s[17]++;
  root.style.setProperty("--initial-color-mode", mode);
  /* istanbul ignore next */
  cov_d6phap09z().s[18]++;
  if (typeof colors === "string") {
    /* istanbul ignore next */
    cov_d6phap09z().b[2][0]++;
    cov_d6phap09z().s[19]++;
    return;
  } else
  /* istanbul ignore next */
  {
    cov_d6phap09z().b[2][1]++;
  }
  cov_d6phap09z().s[20]++;
  Object.entries(colors).forEach(([name, colorByTheme]) =&gt; {
    /* istanbul ignore next */
    cov_d6phap09z().f[3]++;
    cov_d6phap09z().s[21]++;
    root.style.setProperty(variableNameGetter("color", name), colorByTheme[mode]);
  });
})('colorMode', (type, rawName) =&gt; {
  /* istanbul ignore next */
  cov_42pjkn0hx().f[1]++;
  cov_42pjkn0hx().s[4]++;
  return \`--\${type}-\${rawName}\`;
})
</script>
`;
